### PR TITLE
ddl: Remove useless const in DDL worker

### PR DESCRIPTION
**Pull request by JaySon-Huang**
Signed-off-by: JaySon-Huang <tshent@qq.com>

### What problem does this PR solve?

Issue Number: related to https://github.com/pingcap/tidb/issues/25507

Problem Summary:
In [pingcap/parser ddl.go](https://github.com/pingcap/parser/blob/af39ecc734c1f416d5fccf0eed9730df85c78278/model/ddl.go#L84-L88), we export 2 constants `AddIndexStr`,`AddPrimaryKeyStr` strangely. I think it is because TiDB need them in [ddl/ddl_worker.go](https://github.com/pingcap/tidb/blob/9e08fc559c8175d00aefd71fc84667ac4d5dfc21/ddl/ddl_worker.go#L447-L452) to make another DDL worker queue for adding index.

### What is changed and how it works?

Change `newMetaWithQueueTp(txn kv.Transaction, tp string)` to `newMetaWithQueueTp(txn kv.Transaction, tp workerType)` and eliminate the usage of `model.{AddIndexStr,AddPrimaryKeyStr}`. It is also better to compare by a byte but not a string.

After changing it, we can remove the two useless exported const `model.{AddIndexStr,AddPrimaryKeyStr}` from `pingcap/parser`.

### Check List 

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

- No release note

### CODE REVIEW

File: ddl/ddl_worker.go [hunk 1]
```diff
@@ -444,8 +444,8 @@ func isDependencyJobDone(t *meta.Meta, job *model.Job) (bool, error) {
 	return true, nil
 }
 
-func newMetaWithQueueTp(txn kv.Transaction, tp string) *meta.Meta {
-	if tp == model.AddIndexStr || tp == model.AddPrimaryKeyStr {
```
Comment (tisonkun) [inline ddl/ddl_worker.go:448]: Even we lost the reason why `tp == model.AddPrimaryKeyStr` here, LOL.

**Review by tisonkun — APPROVED** [review]
LGTM. We shall later update pingcap/parser code also.

**Review by tangenta — APPROVED** [review]

**Review by zimulala — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
/assign @tangenta @zimulala 

Would you mind to help on merging?

**Pull request merged**